blame.go

```diff
@@ -33,12 +37,6 @@ type BlameResult struct {
 // 2. Then build a graph with a node for every line in every file in
 // the history of the file.
 //
-// Each node (line) holds the commit where it was introduced or
-// last modified. To achieve that we use the FORWARD algorithm
-// described in Zimmermann, et al. "Mining Version Archives for
-// Co-changed Lines", in proceedings of the Mining Software
-// Repositories workshop, Shanghai, May 22-23, 2006.
-//
```

Comment:
Why did you remove this comment? how are you going to understand the current implementation without the paper describing the algorithm? it is probably the most important comment in the whole file.  I agree it should not be public, though, it should be private as it explains the implementation details.